### DISCUSSION TITLE
sudo teleport command not found

### DISCUSSION THREAD

**Discussion by ultratin** [gravitational/teleport#14802] [Q&A] [answered]
I'm trying to stand up teleport to evaluate its capabilities for our org I'm hitting a snag on setting it up. I installed teleport on an EC2 instance running Amazon Linux 2 but I'm getting the following error when trying to run `sudo teleport configure`. Sorry if this is a stupid question. 

```shell
$ sudo teleport configure -o file \
>     --cluster-name=tele.example.com \
>     --public-addr=tele.example.com:443 \
>     --cert-file=/var/lib/teleport/fullchain.pem \
>     --key-file=/var/lib/teleport/privkey.pem
sudo: teleport: command not found
```

**Comment by ultratin** [comment]
I resorted to using the below command to make it work properly. Probably not the best way, if anyone have any other ideas do let me know.

```shell
sudo env PATH="$PATH" teleport configure -o file
```

**Comment by webvictim** [comment, marked as answer]
This is because Amazon Linux 2 doesn't add `/usr/local/bin` to the path when using `sudo` (what's called `secure_path`)

You can run something like this to fix it:

`echo "Defaults    secure_path = /sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin" | sudo tee /etc/sudoers.d/secure_path`

**Reply by llam15** [reply]
Resurrecting an old discussion, but I ran into this issue as well recently. I wanted to raise that the docs at [Database Access with AWS RDS Proxy for PostgreSQL](https://goteleport.com/docs/database-access/guides/rds-proxy-postgres/) have the command listed as 

```bash
# Tip: Add /usr/local/bin to path used by sudo (so 'sudo tctl users add' will work as per the docs)
# echo "Defaults    secure_path = /sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin" > /etc/sudoers.d/secure_path
```

This command fails with `Permission denied`.  Specifically, it seems that `>` fails, whereas `| sudo tee` works.

And just some feedback, the comment mentions `tctl users add`, which is not referenced in this page. This confused me and led me to assume that modifying `/etc/sudoers.d/secure_path` is optional. However, it is a required step, because otherwise you cannot run the later `teleport configure` command. (A bit of a nit, but hopefully will help others avoid this speed bump in the future 😅 )
